### PR TITLE
#486 - Refactoring project structure (3.3.0)

### PR DESCRIPTION
**Pull request by astier**
- Split constructor to several methods
- Remove unnecessary instance variables
- Inline method parameters
- Move getter and setter to the bottom
- Delete unused method parameters
- Change access modifier

### CODE REVIEW

File: webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/detail/AnnotationFeatureForm.java [hunk 1]
```diff
@@ -401,34 +302,184 @@ protected void onConfigure()
...
-        });
-        featureEditorPanel.setOutputMarkupId(true);
+        };
+        selectedAnnotationLayer.setOutputMarkupId(true);
+        return selectedAnnotationLayer;
+    }
 
-        // the annotation layer for the selected annotation
-        selectedAnnotationLayer = new Label("selectedAnnotationLayer", new Model<String>())
+    private Component createDefaultAnnotationLayer()
```
Comment (reckart) [inline webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/detail/AnnotationFeatureForm.java]: The name is misleading - should IMHO contain "selector".

File: webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/detail/AnnotationFeatureForm.java [hunk 2]
```diff
@@ -369,7 +260,7 @@ protected void onUpdate(AjaxRequestTarget aTarget)
                     return;
                 }
                 selectedTag = (forwardAnnotationText.getModelObject() == null ? ""
-                    : forwardAnnotationText.getModelObject().charAt(0)) + selectedTag;
+                        : forwardAnnotationText.getModelObject().charAt(0)) + selectedTag;
```
Comment (astier) [inline webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/detail/AnnotationFeatureForm.java]: Look below at line 376/270.
> aTarget.add(featureEditorPanelContent.get(0));

_org.apache.wicket.Component get(int index)_ is deprecated and marked for deletion for WICKET8.

File: webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/detail/AnnotationFeatureForm.java (hunk 1 above)
Comment (astier) [inline webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/detail/AnnotationFeatureForm.java]: I will rename it to _createDefaultAnnotationLayerSelector_.

**Comment by astier** [comment]
I changed the name from _createDefaultAnnotationLayer_ method and from other methods for better clarity. I also made the following changes:
- Inline method parameters since they are always the same
- Add FIXME comment for deprecated method

**Comment by reckart** [comment]
When you are ready, just squash/set commit message.

**Comment by reckart** [comment]
Ah, do I have to merge #510 before this one?

**Comment by astier** [comment]
No. Don't care about #510. I actually thinking about dropping it. Not sure if it has any benefits since LayerSelector is tightly bounded to AnnotationFeatureForm and ADEP.

**Comment by astier** [comment]
I did few other changes. Check them out and tell me when to make the PR ready for merge.

File: webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/detail/AnnotationDetailEditorPanel.java [hunk 3]
```diff
@@ -111,14 +111,17 @@
 
     private AnnotationFeatureForm annotationFeatureForm;
 
-    public AnnotationDetailEditorPanel(String id, IModel<AnnotatorState> aModel)
+    protected AnnotationDetailEditorPanel(String id, IModel<AnnotatorState> aModel)
```
Comment (reckart) [inline webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/detail/AnnotationDetailEditorPanel.java]: I'd leave the public scope - why not?

File: webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/detail/AnnotationDetailEditorPanel.java [hunk 4]
```diff
@@ -663,12 +666,6 @@ public void actionDelete(AjaxRequestTarget aTarget)
             }
         }
 
-        // If the deleted FS is a relation, we don't have to do anything. Nothing can point to a
```
Comment (reckart) [inline webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/detail/AnnotationDetailEditorPanel.java]: At least the comment should stay. The code/comment don't do anything, but they document that actually nothing *should* be done which is quite helpful.

File: webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/detail/AnnotationDetailEditorPanel.java [hunk 5]
```diff
@@ -776,14 +773,16 @@ public JCas getEditorCas()
         if (state.getMode().equals(Mode.ANNOTATION) || state.getMode().equals(Mode.AUTOMATION)
                 || state.getMode().equals(Mode.CORRECTION)) {
 
+            // FIXME de.tudarmstadt.ukp.clarin.webanno.api.DocumentService readAnnotationCas is
```
Comment (reckart) [inline webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/detail/AnnotationDetailEditorPanel.java]: Do we need a comment for that? The IDE should already be warning about deprecation.

File: webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/detail/AnnotationFeatureForm.java [hunk 6]
```diff
@@ -818,4 +843,29 @@ protected void onUpdate(AjaxRequestTarget aTarget)
             };
         }
     }
+
+    void setSelectedTag(String selectedTag)
```
Comment (reckart) [inline webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/detail/AnnotationFeatureForm.java]: Hm, I personally don't use the package-private scope. Only either public, protected, or private.

**Comment by astier** [comment]
I applied your requests. PR is ready for merge.

**Pull request merged**